dwh/query_lib.py

```diff
@@ -47,14 +50,21 @@ class Runner(Enum):
   #FHIR_SERVER = 3
 
 
-def patient_query_factory(runner: Runner, data_source: str,
-    code_system: str = None) -> PatientQuery:
+def patient_query_factory(
+    runner: Runner,
+    data_source: str,
+    code_system: Optional[str] = None,
+    project_name: Optional[str] = None,
+) -> PatientQuery:
+
   """Returns the right instance of `PatientQuery` based on `data_source`.
 
   Args:
     runner: The runner to use for making data queries
     data_source: The definition of the source, e.g., directory containing
       Parquet files or a BigQuery dataset.
+    project_name: The GoogleCloud project name. This field is required if
```

Comment:
Instead of having this as an explicit argument here, consider adding a `**kwargs` and fail in `_BigQueryPatientQuery` if `project_name` is not provided (also document it in the docstring of `_BigQueryPatientQuery`).